### PR TITLE
Admin merged view with base branch develop-staging

### PR DESCRIPTION
**Pull request by Polej**
To be merged into develop-staging

### CODE REVIEW

File: src/components/Geogebra/GeogebraViews.js [hunk 1]
```diff
@@ -74,15 +75,43 @@ export default class {
             }
         });
 
+        api.service('elements').on('removed', (element) => {
+            const pos = this.workshopIds.indexOf(element.workshop);
```
Comment (BarryOgg) [inline src/components/Geogebra/GeogebraViews.js:79]: `Array` has `includes` method now - we don't need to rely on `indexOf` to determine if it contains an element
https://developer.mozilla.org/pl/docs/Web/JavaScript/Referencje/Obiekty/Array/includes

File: src/components/Geogebra/GeogebraViews.js [hunk 2]
```diff
@@ -74,15 +75,43 @@ export default class {
...
+            if (pos !== -1) {
+                this.GAVs[pos].deleteObject(element.name);
+            }
+        });
+
         api.service('workshops').on('xml-changed', (workshop) => {
             const pos = this.workshopIds.indexOf(workshop.id);
```
Comment (BarryOgg) [inline src/components/Geogebra/GeogebraViews.js:86]: Ditto

File: src/components/Geogebra/GeogebraMergedAdminView.js [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+            );
+
+            this.log.debug('this.workshopIds', this.workshopIds);
+        };
+
+        this.initializeCallbacks = async () => {
+            this.log.debug('Loading initializeCallbacks');
+
+            this.createdListener = (element) => {
+                const pos = this.workshopIds.indexOf(element.workshop);
```
Comment (BarryOgg) [inline src/components/Geogebra/GeogebraMergedAdminView.js:124]: See below - we don't need to use `indexOf` (4 occurences in this file).

**Review by BarryOgg — CHANGES_REQUESTED** [review]

File: src/components/Geogebra/GeogebraViews.js (hunk 1 above)
Comment (Polej) [reply src/components/Geogebra/GeogebraViews.js:79]: I need the position in array, not only Boolean.

File: src/components/Geogebra/GeogebraMergedAdminView.js (hunk 3 above)
Comment (Polej) [reply src/components/Geogebra/GeogebraMergedAdminView.js]: I need the position in array, not only Boolean.

File: src/components/Geogebra/GeogebraViews.js (hunk 2 above)
Comment (Polej) [reply src/components/Geogebra/GeogebraViews.js]: I need the position in array, not only Boolean.

Comment (Polej) [reply src/components/Geogebra/GeogebraViews.js:86]: I need the position in array, not only Boolean.

File: src/components/Geogebra/GeogebraMergedAdminView.js (hunk 3 above)
Comment (Polej) [reply src/components/Geogebra/GeogebraMergedAdminView.js:124]: I need the position in array, not only Boolean.

**Review by BarryOgg — APPROVED** [review]

**Pull request merged**